tutorials/shading/your_first_shader/what_are_shaders.rst

```diff
@@ -0,0 +1,111 @@
+.. _doc_what_are_shaders:
+
+What are shaders?
+=================
+
+Introduction
+------------
+
+So, you have decided to give shaders a try. You have likely heard that they can be used to 
+create interesting effects that run incredibly fast. You have also likely heard that they
+are terrifying. Both are true. 
+
+Shaders can be used to create a wide range of effects (in fact everything drawn in a modern
+rendering engine is done with shaders). 
+
+Writing shaders can also be very difficult for people unfamiliar with them. Godot tries to make writing
+shaders a little easier by exposing many useful built-in features and handling some of the
+lower-level initialization work for you. However, GLSL (the OpenGL Shading Language, which Godot uses)
+is still unintuitive and restricting, especially for users who are used to GDScript.
+
+But what are they?
+------------------
+
+Shaders are a special kind of program that runs on Graphics Processing Units (GPUs). All computers
+have some sort of GPU, either one integrated into their CPU or discrete (meaning it is a separate 
+hardware component). GPUs are especially useful for rendering because they are optimized for running
+thousands of instructions at once.
+
+The output of the shader is the colored pixels of the object drawn to the viewport. Shaders operate
```

Comment:
>The output of the shader is the colored pixels of the object drawn to the viewport

This is true for many typical graphics shaders, but it doesn't need to be colored (think greyscale, or black and white), and it doesn't need to be pixels - think OpenCL/CUDA calculations etc.

Reply:
One the one hand, its nice to keep it simple as this is meant to go along with with "your first shader" tutorials. On the other hand it is better to be more correct. I added "typically" then also added a line saying that other outputs are possible.